kwok/cloudprovider/cloudprovider.go

```diff
@@ -229,12 +229,26 @@ func (c CloudProvider) toNode(nodeClaim *v1.NodeClaim) (*corev1.Node, error) {
 			// so we only apply resource requests, since NodeOverlay will not apply.
 			// If this changes in the future, we'll need to update capacity and allocatable values for KWOK nodes.
 			Capacity:    nodeClaim.Spec.Resources.Requests,
-			Allocatable: lo.Assign(nodeClaim.Spec.Resources.Requests, instanceType.Allocatable()),
+			Allocatable: minResourceList(nodeClaim.Spec.Resources.Requests, instanceType.Allocatable()),
```

Comment:
I'm not deep on this code, but it strikes me as odd that we're doing different thinks for Capacity and Allocatable here. Why is the logic different?